[Release] 1.5.0

JoeMatt opened the pull request:
Release branch cut from current #master since it has commits after #534 .

To be correct, a new release should be cut before merging more tickets into `develop` so that `master` can represent the current release state as per `git-flow` guidelines.

All other PRs have been changed to merge against `develop`

JoeMatt commented:
GitHub in my experience messes up git-flow since develop should merge from master next to maintain the tag on both branches.

It works with `git flow release finish`, so not sure how to handle it in this case.

In my other projects, we unlock master & develop, push the release merges to both branches, then relock.

Thoughts?

kuyazee commented:
I agree with this, @JoeMatt since technically we're the maintainers now as stated by @lkzhao on discord, I think we should also go with this since it's cleaner and safer to work this way.

JoeMatt commented:
I'm actually stuck without elevated access. Maintainers have access to the settings panel on GitHub projects, I do not. 

1. Branch locking needs to be configured
2. Something needs to be done to publish update public podspec to cocoapods

For 2 I think there's an avenue to do that, but would be better if the current admin hands it over.

kuyazee commented:
> I'm actually stuck without elevated access. Maintainers have access to the settings panel on GitHub projects, I do not.
> 
> 1. Branch locking needs to be configured
> 2. Something needs to be done to publish update public podspec to cocoapods
> 
> For 2 I think there's an avenue to do that, but would be better if the current admin hands it over.

Ah yes I also have this problem for the other repository I'm maintaining. Especially with publishing the updated podspec to cocoapods. It would be great if we could automate it on Travis but authenticating the trunk always needs to go by email. 

Have any ideas besides us having the ff:
- Admin rights on the repository
- Access to the attached email for `Hero`

lkzhao commented:
Thanks for putting this up. Let me try if I can give you guys admin access if possible. As for cocoapods, I can add you guys as owner

JoeMatt commented:
Sounds good. 

Might be useful to make a fastlane action to do releases.

I think the steps are basically,

1. Run tests
2. Version bump (major, minor, bug)
3. Check/Update changelog (or auto-gen)
4. Change the version number in Podspec, .jazzy.yaml
5. Run jazzy
6. Podspec lint
7. git commit
8. git flow release finish
9. git push
10. Publish Cocoapods spec
11. *Optional* carthage archive --platform iOS
12. Make github release from version tag (attach carthage archive zip)

Pretty much all of these have some included fastlane action so a good candidate.

JoeMatt commented:
Just wanted to update, I have a few CI things to finish setting up.

I'll be making a PR with some config updates first, then finish this release.

Planning for this weekend Oct 12-13. Coordinate on Discord later.

JoeMatt commented:
I cherry-picked some commit from develop that were essential to fix breaks in XCode 10 for the demo app and unit tests, and fixed pod lib lint warnings and updated the pod spec's metadata a bit.

If this is all good then this release is ready for merge.

iDevid commented:
Changelog is not updated.

kuyazee commented on Hero.podspec:
```diff
@@ -1,24 +1,36 @@
+  s.ios.frameworks = 'UIKit', 'Foundation', 'QuartzCore', 'CoreGraphics', 'CoreMedia'
+  s.tvos.frameworks = 'UIKit', 'Foundation', 'QuartzCore', 'CoreGraphics', 'CoreMedia'
+
+  s.swift_version = '4.2'
```
We're already using Swift 5 for `1.5.0` release right?

JoeMatt replied:
No, develop is, this release branch was cut before those merges due to `master` having commits after 1.4.0.

1.6.0 will have swift 5+.

kuyazee replied:
Oh ok.

JoeMatt commented:
> Changelog is not updated.

Yes it is, 

https://github.com/HeroTransitions/Hero/pull/620/files#diff-3
